data/hydraulic-data.py

```diff
@@ -0,0 +1,129 @@
+"""Create hydraulic systems dataset"""
+
+from typing import Union, List
+from h2oaicore.data import CustomData
+import datatable as dt
+import numpy as np
+import pandas as pd
+
+class HydraulicSystemsData(CustomData):
+    # url contains hydraulic systems condition monitoring data set
+    base_url = "https://archive.ics.uci.edu/ml/machine-learning-databases/00447/"
+
+    @staticmethod
+    def create_data(X: dt.Frame = None) -> Union[str, List[str],
+                                                 dt.Frame, List[dt.Frame],
+                                                 np.ndarray, List[np.ndarray],
+                                                 pd.DataFrame, List[pd.DataFrame]]:
+        import os
+        from h2oaicore.systemutils_more import download
+        from h2oaicore.systemutils import config
+        import zipfile
+
+        def mean_conversion(pd_frame):
+            """
+                Means are computed along columns to average the cycle sensor data
+                Each row in frame is data collected over a cycle of 60 seconds
+            """
+            return pd_frame.mean(axis = 1) # axis=1 means along column
+
+        def convert_to_frame(file_bytes_object):
+            """
+                Convert bytes data to pandas dataframe
+            """
+            data_file_str = str(file_bytes_object,"utf-8")
+            dt_frame = dt.Frame(data_file_str)
+            pd_frame = dt_frame.to_pandas()
+            return pd_frame
+
+        def import_data_field(zip, filename, fieldname):
+            """
+                Loads pandas dataframe with average cycle sensor data
+            """
+            pd_frame = pd.DataFrame( mean_conversion(convert_to_frame(zip.read(filename))) )
+            pd_frame.columns = [fieldname]
+            return pd_frame
+
+        def import_target_label(zip, filename, target_label):
+            """
+                Loads pandas dataframe with target label field
+            """
+            pd_frame = convert_to_frame(zip.read(filename))
+            if target_label == "cool_cond_y":
+                # store hydraulic cooler condition label data from column 0 into pd frame
+                pd_targetLabelY = pd.DataFrame(pd_frame.iloc[:,0])
```

Comment:
Just adding a note here to reference what we talked about regarding using a dictionary instead of a series of if/elif.

Reply:
Thanks for your feedback! I replaced the series of if/elif statements to a dictionary. The code is much easier to read now